src/main/app/MiningFramework.groovy

```diff
@@ -163,16 +180,23 @@ class MiningFramework {
     }
 
     private void cloneRepository(Project project, String target) {
-
+        
         println "Cloning repository ${project.getName()} into ${target}"
 
         File projectDirectory = new File(target)
         if(projectDirectory.exists()) {
             FileManager.delete(projectDirectory)
         }
         projectDirectory.mkdirs()
+        
+        String url = project.getPath()
+        String token = arguments.getAccessKey();
+        if (token.length() > 0) {
```

Comment:
Essa checagem deveria ser responsabilidade do parser de argumentos, não?
Fora isso, além dessa estratégia só ter utilidade quando o projeto for privado, força que todos os projetos sejam de acesso do mesmo desenvolvedor, já que é uma senha para todos da lista. Acho que o programa deveria perguntar a senha quando identificar que o projeto é privado ou passar o controle para o Git para que ele pergunte.

Reply:
A análise precisa fazer o clone assim, por que em sistemas Unix, toda vez que você precisa dar push, ele pedirá usuário e senha do github. Como para coletar as builds é feito um push no fork para cada commit, é necessária uma forma de pular essa autenticação, ao usar o clone o push  vai direto.

Reply:
Entendi, realmente. Ainda tem uma alternativa de o programa fornecer o usuário e a senha automaticamente, mas isso é melhor deixar pra outra issue, se for o caso.